### DISCUSSION TITLE
Using a socks5 proxy

### DISCUSSION THREAD

**Discussion by oculunto** [seleniumbase/SeleniumBase#2518] [Q&A] [answered]
Hello, I'm trying to use a socks5 proxy with Driver, but I stumble across "ERR_NO_SUPPORTED_PROXIES" error. 
I am initializing the Driver like this:

`browser = Driver(uc=True,binary_location="C:\\Program Files\\Google\\Chrome Beta\\Application\\chrome.exe",page_load_strategy="none",user_data_dir = profileDir, proxy="socks5://"+PROXY_HOST+":"+PROXY_PORT) `

I've worked with normal proxies so far and they work fine, but I was trying to get into socks5 ones. What am I doing wrong ?

**Comment by mdmintz** [comment, marked as answer]
If you're trying to do SOCKS5 proxy **with authentication**, there's info on that here: https://github.com/seleniumbase/SeleniumBase/issues/931#issuecomment-875201915

The closest solution I've found is https://stackoverflow.com/a/55395796 , which uses https://github.com/qwj/python-proxy . There's no direct way to do it, as seen from https://stackoverflow.com/questions/43312573/how-to-set-socks5-proxy-with-auth-for-chrome-in-selenium-python

----

If you're trying to do SOCKS5 proxy **without authentication**, then it works if you remove setting a `user_data_dir`. If you must use a `user_data_dir` in UC Mode with special Proxy settings, then make sure that **every** Chrome instance that uses that `user_data_dir` has also used it with UC Mode with the **exact same proxy settings**. This means that if a regular Chrome (non-UC-Mode) touched that `user_data_dir`, then you can't use it with UC Mode anymore. Also, if a Chrome with no proxy settings or a Chrome with different proxy settings touch that `user_data_dir`, then you can't change the proxy settings for that `user_data_dir` anymore. In those cases, set a new `user_data_dir` to use when you run your script next.

----

Also, (in case it's the actual issue), make sure you're using valid proxy settings. If you use an invalid proxy server (or set proxy settings incorrectly), then you'll run into issues. Your SOCKS5 proxy settings should match this format:

```python
"socks5://IP_ADDRESS:PORT"
```

**Reply by oculunto** [reply]
Thank you, deleting `user_data_dir` made it work ! (although I was using it with new directories before, since I delete the dir everytime I switch proxies). 

Another question if you don't mind, is it intentional not to use http3 in uc mode or is it a proxy problem on my end ? I am trying to avoid being detected by the fact that I am limiting my requests to http2 (avoiding QUIC ?). 

Thanks for the assistance and for your great work !

**Reply by mdmintz** [reply]
I wouldn’t know which changes to make specifically for https://en.m.wikipedia.org/wiki/HTTP/3 yet.

**Reply by oculunto** [reply]
So it is a legitimate issue, detection by blocking http3/only using http2 ?

**Reply by mdmintz** [reply]
I haven't seen anything like that in the original `undetected-chromedriver` repo.
